### PR TITLE
mimxrt10xx: implement i2sout

### PR DESCRIPTION
**Pull request by jepler**
mimxrt10xx: implement i2sout

tested on metro m7 (green prototype version) with max98357a i2s amplifier and the following test code:
```py
import board
import time
import digitalio
from audiobusio import I2SOut
from audiocore import RawSample
from microcontroller import pin
from ulab import numpy as np

n = np.array(np.sin(np.linspace(0, np.pi*2, 218, endpoint=False)) * 200, dtype=np.int16)
print(n)
r = RawSample(n, sample_rate=8000, channel_count=2)
def main():
    with digitalio.DigitalInOut(board.LED) as l:
        l.switch_to_output(True)
        value = False
        while True:
            with I2SOut(pin.GPIO_06, pin.GPIO_07, pin.GPIO_04) as i:
                time.sleep(.01)
                l.value = value = not value
                i.play(r, loop=True)
                print(i.playing)
                time.sleep(.5)
                i.stop()
                print("STOPPED")
                print(i.playing)
                time.sleep(.5)
                i.play(r, loop=True)
                print(i.playing)
                print("PLAY AGAIN")
                time.sleep(.5)
            time.sleep(1)
```

Tested (all on metro M7 using D9/10/12 with max98357a i2s amplifier):
 * various sample rates (not calibrated, but a familiar song didn't sound off-key)
 * stereo/mono
 * 8/16 bit
 * mp3 playback
 * mixer playback (with up to ~12, 22kHz, mono, 16-bit channels)

As expected, it stutters when flash writes occur (e.g., by the host computer) but contrary to earlier testing it is not crashing or no longer responding to ctrl-c; this may have been due to the earlier tests having copious prints in the background callback function, all the debugging. Stutters are short dropouts, not horrible brain-rending static, so that's fine.

### CODE REVIEW

**Comment by jepler** [comment]
it needs the tables of i2s pins written for the other micros

File: ports/mimxrt10xx/common-hal/audiobusio/__init__.c [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+
+                case SAMPLE_TYPE(false, 1, 1):
+                    audiosample_convert_u8m_s16s((int16_t *)ptr, (uint8_t *)(void *)self->sample_data, framecount);
+                    break;
+            }
+            self->sample_data += bytes_per_input_frame * framecount; // in bytes
+            ptr += framecount; // in frames
+        }
+        // Fill any remaining portion of the buffer with 'no sound'
+        memset(ptr, 0, (end - ptr) * sizeof(uint32_t));
```
Comment (tannewt) [inline ports/mimxrt10xx/common-hal/audiobusio/__init__.c:278]: Don't we allow the user to set the quiescent value? Or is it ok that it is always signed?

**Review by tannewt — CHANGES_REQUESTED** [review]
Two questions. Looks good otherwise!

File: ports/mimxrt10xx/background.c [hunk 2]
```diff
@@ -36,7 +36,7 @@ void PLACE_IN_ITCM(port_background_task)(void) {
 
 void port_background_tick(void) {
     #if CIRCUITPY_AUDIOIO || CIRCUITPY_AUDIOBUSIO
-    audio_dma_background();
+    // audio_dma_background();
     #endif
```
[the remark this answers is not in the document]
Comment (jepler) [reply ports/mimxrt10xx/background.c]: Done

**Review by tannewt — APPROVED** [review]
Thank you!

**Pull request merged**